datacube/utils/geometry/_base.py

```diff
@@ -857,6 +857,38 @@ def __getitem__(self, roi):
 
         return GeoBox(width=w, height=h, affine=affine, crs=self.crs)
 
+    def __or__(self, other):
```

Comment:
what about different resolutions, what's output in this case, is it defined, does it pick highest resolution or lowest needs more documentation than this.

Separate issue: if we provide OR we must also provide AND, otherwise it's confusing.